### PR TITLE
🐛 Fix/storage API backwards compatibility with deprecated storage client sdk

### PR DESCRIPTION
**Pull request by pcrespov**
**TL;DR** 
Legacy services cannot download data due to incompatibilities with the latest storage OAS.

## What do these changes do?

Legacy services use different versions of ``simcore-sdk`` library to perform upload/download of data. 
Old versions of ``simcore-sdk`` use a deprecated client sdk for storage called ``simcore_service_storage_sdk`` that was built against OAS 0.1.0 and current version of the storage OAS is not fully compatible

Here is an example with legacy jupyter-math. The data is actually uploaded but when closed and re-opened the state is not recovered:
before

In addition, I did update [services/storage api version: 0.2.1 → 0.3.0](https://github.com/ITISFoundation/osparc-simcore/pull/3198/commits/67764b5b6e5077ad52fbf3c8a88c97fabbae71cb) since it was forgotten last time we the changed OAS

## How to test

Repeated same manual test as above after adding the fix
after

CI: ``services/storage/tests/unit/test__legacy_storage_sdk_compatibility.py``

## Checklist

- [x] Openapi changes? ``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x ] Documentation reflects the changes

### CODE REVIEW

File: services/storage/requirements/_test.in [hunk 1]
```diff
@@ -5,6 +5,8 @@
 --constraint ../../../requirements/constraints.txt
 --constraint _base.txt
 
+# to test backwards compatibility against deprecated client-sdk (used still in old versions of simcore-sdk)
+simcore-service-storage-sdk @ git+https://github.com/ITISFoundation/osparc-simcore.git@cfdf4f86d844ebb362f4f39e9c6571d561b72897#subdirectory=services/storage/client-sdk/python
```
Comment (GitHK) [inline services/storage/requirements/_test.in:9]: I see you install the SDK that the legacy services use. This way you can test with their interface agains the new API. Great.

**Review by GitHK — APPROVED** [review]
🎉 thanks for this! Just one minor suggestion below.

File: services/storage/src/simcore_service_storage/handlers_files.py [hunk 2]
```diff
@@ -91,6 +91,29 @@ async def get_file_metadata(request: web.Request):
         # TODO: once all legacy services are gone, remove the try except, it will default to 404
         return {"error": "No result found", "data": {}}
 
+    if request.headers.get("User-Agent") == "OpenAPI-Generator/0.1.0/python":
+        # LEGACY compatiblity with API v0.1.0
+        # SEE models used in sdk in:
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/storage/src/simcore_service_storage/handlers_files.py:96]: it is already added in the test since that is the one covering the steps that reproduce that bug. 
That change is mostly related with the models in the client sdk and there are already links for those there

**Review by mrnicegyu11 — APPROVED** [review]
gtm thanks for this
